### ISSUE TITLE
Nodes and requires_grad=True (pytorch optimization within comfyui)

### ISSUE THREAD

**Issue by jloveric** [comfyanonymous/ComfyUI#2946]
I have a node that does some pytorch optimization, simple regression over coordinates. When I run this outside of ComfyUI it works fine, inside all the gradient information is missing and so cannot optimize. Is there something in comfyUI that is disabling all the pytorch gradients? For example with something like set_grad_enabled(False)? At any rate, even using that function and setting to True within a context manager does not resolve the problem.

**Comment by jloveric** [comment]
Ok, using the context manager with at the node level does set is_grad_enabled=True in child functions
 with torch.set_grad_enabled(True) though still running into the same issue.

**Comment by comfyanonymous** [comment]
ComfyUI has torch.inference_mode turned on, you can turn it off in your node if needed by doing: with torch.inference_mode(False)

https://pytorch.org/docs/stable/generated/torch.inference_mode.html
